File: Data Structures & Algorithms/contains-duplicate-ii/submission-4.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        h = set()
        L = 0
        for R in range(len(nums)):
            if R - L > k:
                h.remove(nums[L])
                L+=1
            if nums[R] in h:
                return True
            h.add(nums[R])
        return False

File: Data Structures & Algorithms/binary-search/submission-4.java
class Solution {
    public int search(int[] nums, int target) {
        int L = 0;
        int R = nums.length - 1;
        int middle = (L + R) / 2;

        while (L <= R) {
            if (nums[middle] < target) {
                L = middle + 1;
                middle = (L + R) / 2;
            } else if (nums[middle] > target) {
                R = middle - 1;
                middle = (L + R) / 2;
            } else {
                return middle;
            }
        }

        return -1;
    }
}
